app/config/baffs.py

```diff
@@ -0,0 +1,12 @@
+class Buffs:
+    def __init__(self, buffs: dict):
+        self.buffs = buffs
+
+    def take_buff(self, knight) -> None:
+        if self.buffs is not None:

```

Comment:
These code duplications can be fixed, using `stats = ["protection", "power", "hp"]` and looping on `stats`